jdaviz/core/data_formats.py

```diff
@@ -112,3 +122,145 @@ def identify_data(filename, current=None):
+        CCDData
+    ]
+
+    if filename.lower().endswith('asdf'):
+        # ASDF files are only supported in jdaviz for
+        # Roman WFI 2D images, so suggest imviz:
+        return 'imviz'
+
+    header = fits.getheader(filename, ext=ext)
+    data = fits.getdata(filename, ext=ext)
+    wcs = _get_wcs(filename, header)
+    has_spectral_axis = 'spectral' in wcs.world_axis_object_classes
+
+    n_axes = (
+        int(has_spectral_axis) +
+
+        sum([component[0] in ['celestial', 'angle']
+             for component in wcs.world_axis_object_components]) -
+
+        # remove any slit_frame axis from the count
+        (0 if not hasattr(wcs, 'available_frames') else
+         int('slit_frame' in wcs.available_frames))
+    )
+
+    # use astropy to recognize some data formats:
+    possible_formats = {}
+    for cls in supported_dtypes:
+        fmt = registry.identify_format(
+            'read', cls, filename, None, {}, {}
+        )
+        if fmt:
+            possible_formats[cls] = fmt
+
+    # If CCDData is the only match:
+    if len(possible_formats) == 1:
+        only_key, only_value = possible_formats.popitem()
+        if only_key == CCDData:
+            # could be 2D spectrum or 2D image. break tie with WCS:
+            if has_spectral_axis:
+                if n_axes > 1:
+                    return 'specviz2d'
+                return 'specviz'
+            elif not isinstance(data, fits.BinTableHDU):
+                return 'imviz'
+
+    # Ensure specviz is chosen when ``data`` is a table or recarray
+    # and there's a "known" spectral column name:
+    if isinstance(data, (fits.BinTableHDU, fits.fitsrec.FITS_rec)):
+        # now catch spectra in FITS tables, looking for
+        # columns with "wave" or "flux" in the names:
+        table_columns = [getattr(col, 'name', col).lower() for col in data.columns]
+
+        # these are "known" prefixes for column names
+        # in FITS tables of spectral observations
+        known_spectral_columns = [
+            'wave',
+            'flux'
+        ]
+
+        # this list of bools indicates any
+        # spectral column names found:
+        found_spectral_columns = [
+            found_col.startswith(known_col)
+            for known_col in known_spectral_columns
+            for found_col in table_columns
+        ]
+
+        # if at least one spectral column is found:
+        if sum(found_spectral_columns):
+            return 'specviz'
+
+    # If the data could be spectral:
+    for cls in [Spectrum1D, SpectrumList]:
+        if cls in possible_formats.keys():
+            recognized_spectrum_format = possible_formats[cls][0].lower()
+
+            # first catch known JWST spectrum types:
+            if (n_axes == 3 and
+                    recognized_spectrum_format.find('s3d') > -1):
+                return 'cubeviz'
+            elif (n_axes == 2 and
+                  recognized_spectrum_format.find('x1d') > -1):
+                return 'specviz'
```

Comment:
I wonder if it's worth generalizing these now to something non-JWST specific.  For example, this function already fails with MaNGA cubes files, which is maybe a tad bittersweet since Jdaviz was originally developed against those files.  :(  And for MAST to switch to this function, we'd need this generalized before we can start any work on our end.

Reply:
I agree that there's a better way to do this! Stepping away for a few days and seeing this comment makes it clearer.

This PR relies on `astropy.io.registry` to auto-identify file types, and then when that fails, uses heuristics to make a guess. The astropy registry knows some JWST files without a problem, so I didn't _need_ to use something else. 

_But_ there is already a registry which identifies JWST, MaNGA, and many others: the [specutils registry](https://github.com/astropy/specutils/blob/2f44758450e79315cb2516f56632174245dc42f9/specutils/io/default_loaders/jwst_reader.py#L19). The easy solution to all this is to also try the specutils registry. I'll do this today! 🤦🏻

Reply:
Support for MaNGA introduced in [7b8d713](https://github.com/spacetelescope/jdaviz/pull/2124/commits/7b8d71343e2717730e4dba59f86e2e1b8f589975).

Reply:
Thanks for the update.  The MaNGA cube and rss files do seem to now correctly identify as cubeviz and specviz, which is great.  Since JWST formats are covered by the `identify_data` function, it's not clear to me why we're still explicitly checking on the filename suffix at lines 241-246?